### PR TITLE
Convert instructor assessment regrading and upload pages to async

### PR DESCRIPTION
**Pull request by mylesw27**
This PR intends to convert the `instructorAssessmentRegrading` and `instructorAssessmentUploads` pages to async as well as use Zod-validated queries.

### CODE REVIEW

File: apps/prairielearn/src/lib/db-types.ts [hunk 1]
```diff
@@ -924,3 +924,20 @@ export const GradingJobStatusSchema = z.enum([
...
+  course_request_id: IdSchema.nullable(),
+  description: z.string().nullable(),
+  finish_date: DateFromISOString.nullable(),
+  id: IdSchema,
+  legacy: z.boolean(),
+  number: z.number(),
+  start_date: DateFromISOString,
+  status: z.string(),
+  type: z.string(),
+  user_id: IdSchema,
```
Comment (nwalters512) [inline apps/prairielearn/src/lib/db-types.ts:942]: Many of these columns are nullable (all those without a `NOT NULL` constraint).

Comment (mylesw27) [reply apps/prairielearn/src/lib/db-types.ts]: Updated nullable columns in [448eb1b](https://github.com/PrairieLearn/PrairieLearn/pull/9734/commits/448eb1b6548ad7de731de1737fe2c7995d0badff).

File: apps/prairielearn/src/lib/db-types.ts [hunk 2]
```diff
@@ -924,3 +924,20 @@ export const GradingJobStatusSchema = z.enum([
...
+  course_id: IdSchema.nullable(),
+  course_instance_id: IdSchema.nullable(),
+  course_request_id: IdSchema.nullable(),
+  description: z.string().nullable(),
+  finish_date: DateFromISOString.nullable(),
+  id: IdSchema,
+  legacy: z.boolean(),
+  number: z.number().nullable(),
+  start_date: DateFromISOString.nullable(),
+  status: z.string().nullable(),
```
Comment (nwalters512) [inline apps/prairielearn/src/lib/db-types.ts:940]: This is actually an enum column, and in this case, we prefer to use an enum type, mostly for IDE completion.

For an example, see `EnumPlanGrantTypeSchema` here:

https://github.com/PrairieLearn/PrairieLearn/blob/ccf6649762c1c73ceeb2aa17efd686eca0d0df41/apps/prairielearn/src/lib/db-types.ts#L325

This is then used in `PlanGrantSchema` for the `type` column:

https://github.com/PrairieLearn/PrairieLearn/blob/ccf6649762c1c73ceeb2aa17efd686eca0d0df41/apps/prairielearn/src/lib/db-types.ts#L334

Comment (mylesw27) [reply apps/prairielearn/src/lib/db-types.ts]: Added `EnumJobStatusSchema` in [eaded80](https://github.com/PrairieLearn/PrairieLearn/pull/9734/commits/eaded80aaf1048e91535830298092335273e5901). Now using it here and in `JobSchema.status`.

**Review by nwalters512 — APPROVED** [review]

**Pull request merged**